service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreatePrivateEndpointStep.java

```diff
@@ -0,0 +1,96 @@
+package bio.terra.landingzone.stairway.flight.create.resource.step;
+
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.library.landingzones.definition.ResourceNameGenerator;
+import bio.terra.landingzone.library.landingzones.definition.factories.CromwellBaseResourcesFactory;
+import bio.terra.landingzone.library.landingzones.definition.factories.ParametersResolver;
+import bio.terra.landingzone.service.landingzone.azure.model.LandingZoneResource;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.network.models.PrivateLinkSubResourceName;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreatePrivateEndpointStep extends BaseResourceCreateStep {
+  private static final Logger logger = LoggerFactory.getLogger(CreatePrivateEndpointStep.class);
+  public static final String PRIVATE_ENDPOINT_ID = "PRIVATE_ENDPOINT_ID";
+  public static final String PRIVATE_ENDPOINT_RESOURCE_KEY = "PRIVATE_ENDPOINT";
+
+  public CreatePrivateEndpointStep(
+      ArmManagers armManagers,
+      ParametersResolver parametersResolver,
+      ResourceNameGenerator resourceNameGenerator) {
+    super(armManagers, parametersResolver, resourceNameGenerator);
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    var privateEndpointId = context.getWorkingMap().get(PRIVATE_ENDPOINT_ID, String.class);
+    try {
+      if (privateEndpointId != null) {
+        armManagers.azureResourceManager().privateEndpoints().deleteById(privateEndpointId);
+        logger.info("{} resource with id={} deleted.", getResourceType(), privateEndpointId);
+      }
+    } catch (ManagementException e) {
+      if (StringUtils.equalsIgnoreCase(e.getValue().getCode(), "ResourceNotFound")) {
```

Comment:
To be consistent with the other steps., consider adding a logger here indicating that the resource was already deleted.